File: Merge Two Sorted Lists - Leetcode 21/Merge Two Sorted Lists - Leetcode 21.ts
class ListNode {
  public val: number;
  public next: ListNode | null;
  constructor(val?: number, next?: ListNode) {
    this.val = val === undefined ? 0 : val;
    this.next = next === undefined ? null : next;
  }
}

const mergeTwoLists = (list1: ListNode | null, list2: ListNode | null) => {
  const head = new ListNode();
  let cur = head;

  while (list1 !== null && list2 !== null) {
    if (list1.val < list2.val) {
      cur.next = list1;
      list1 = list1.next;
    } else {
      cur.next = list2;
      list2 = list2.next;
    }
    cur = cur.next;
  }

  cur.next = list1 ? list1 : list2;

  return head.next;
};

File: Merge Two Sorted Lists - Leetcode 21/Merge Two Sorted Lists - Leetcode 21.py
class ListNode:
    def __init__(self, val=0,next=None):
        self.val = val
        self.next = next

def mergeTwoLists(list1: ListNode, list2:ListNode) -> ListNode:
    head = ListNode()
    cur = head

    while list1 and list2:
        if list1.val < list2.val:
            cur.next = list1
            lis1 = list1.next
        else:
            cur.next = list2
            list2 = list2.next
        cur = cur.next
    
    cur.next = list1 if list1 else list2

    return head.next
        